networking/net.go

```diff
@@ -49,23 +41,37 @@ func LoadNets() ([]Net, error) {
 		return nil, err
 	}
 
-	var nets []Net
-
+	files := []string{}
 	for _, dent := range dirents {
 		if dent.IsDir() {
 			continue
 		}
 
-		nf := path.Join(RktNetPath, dent.Name())
+		files = append(files, dent.Name())
+	}
+
+	return files, nil
+}
+
+func LoadNets() ([]Net, error) {
+	files, err := listFiles(RktNetPath)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Strings(files)
+
+	var nets []Net
```

Comment:
+1, I'm also trying to break the habit and make that same switch
On Jan 15, 2015 9:18 PM, "Eugene Yakubovich" <notifications@github.com>
wrote:

> In networking/net.go
> <https://github.com/coreos/rocket/pull/385#discussion-diff-23063660>:
>
> > -		nf := path.Join(RktNetPath, dent.Name())
> > +		files = append(files, dent.Name())
> > +	}
> > +
> > +	return files, nil
> > +}
> > +
> > +func LoadNets() ([]Net, error) {
> > +	files, err := listFiles(RktNetPath)
> > +	if err != nil {
> > +		return nil, err
> > +	}
> > +
> > +	sort.Strings(files)
> > +
> > +	var nets []Net
>
> I'm trying to change from doing var x []string to x := []string{} as the
> latter seems more popular.
> Good point on the prealloc.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rocket/pull/385/files#r23063660>.
>